query/src/sql/planner/semantic/type_check.rs

```diff
@@ -446,9 +451,12 @@ impl<'a> TypeChecker<'a> {
     pub async fn resolve_function(
         &mut self,
         func_name: &str,
-        arguments: &[&Expr<'a>],
+        arguments: &[&Expr<'_>],
         _required_type: Option<DataTypeImpl>,
     ) -> Result<(Scalar, DataTypeImpl)> {
+        if !is_builtin_function(func_name) {
+            return self.resolve_udf(func_name, arguments).await;
+        }
```

Comment:
We should move this check to https://github.com/datafuselabs/databend/blob/83884cd152d43c5d77907262a4a58f2d8e0be083/query/src/sql/planner/semantic/type_check.rs#L259-L268

Reply:
done